### DISCUSSION TITLE
Ignoring folders

### DISCUSSION THREAD

**Discussion by Ronald1302** [photoprism/photoprism#2077] [Q&A] [answered]
I have my photo's organized in such a way that my main folder contains all the unedited files, and I export all my edited and selected files to a subfolder called export. From what I've found so far, the .ppignore file will also ignore all files in subdirectories. Is there a way to specify specifically which folder to include instead of only which to exclude?

In short, I want to include only the following italic folders, and ignore all photo's in the underlaying directories:

- 2021
- - 01 event
- - - _export_
- - 02 event
- - - _export_

(I hope I'm clear, my actual folder structure is more complex, so I don't know how much value a screenshot would add...)

**Comment by lastzero** [comment, marked as answer]
Not yet, but you could mount only the folders you want to index...

**Reply by vanderneut** [reply]
On my Synology I have PhotoPrism pointed to my /photo folder and it works, but it also scans through the /photo/#recycle sub-folder. I would love to be able to specify to ignore any folder by the name "#recycle". To instead mount all the sub-folders under /photo that I do want to index is not really a feasible workaround, as that is a continuously growing set of folders. I really just want to be able to say: "Index anything under /photo, *except* anything under /photo/#recycle."

**Reply by graciousgrey** [reply]
To exclude files or folders from being indexed you can use a .ppignore file. It must be added before initial indexing, see: https://docs.photoprism.app/user-guide/library/originals/#ignoring-files-and-folders

**Reply by charles-cai** [reply]
But I don't want the default behavior: "Hidden files and folders that start with a ., @, _., or __ like __MACOSX are automatically ignored..."

Is there a way to edit the above rules to allow my folders already started with __ (for example, "_to_be_processed") to be INCLUDED in the indexing/scanning?  I really need this as my workflow and naming convention uses _ underscore a lot, but don't want them to be ignored by PhotoPrism scan. 

Thanks
Charles

**Reply by graciousgrey** [reply]
Unfortunately there is no config option for these rules available at this time.

**Reply by lastzero** [reply]
I don't believe a single _ should be a problem?

**Comment by gadelkareem** [comment]
@lastzero would something like this work for /volume/photos/#snapshot and /volume/photos/@eaDir
```
[#]*
[#]*/*
[#]*/*/*
[#]*/*/*/*
[#]*/*/*/*/*
[#]*/*/*/*/*/*
[#]*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[#]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*
[@]*/*
[@]*/*/*
[@]*/*/*/*
[@]*/*/*/*/*
[@]*/*/*/*/*/*
[@]*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*
[@]*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*/*

````

**Reply by lastzero** [reply]
Is that a Christmas tree? 🎄

**Reply by lastzero** [reply]
Joking aside: If you add `[#]*` to the `.ppignore` file in the root of your *originals* folder, all file and folders starting with `#` should be ignored. Folders starting with `@` should be skipped anyway. If unsure, I suggest you try this with a few sample files.

Also, we could exclude the special directories that Synology uses by default, so that no manual configuration is required. For this, it would be good to have a list or a specification.

**Reply by gadelkareem** [reply]
Understood, thank you! I was under the impression that subfolders weren't ignored because they kept appearing in the index. Therefore, I had to remove everything and begin anew.

**Reply by lastzero** [reply]
So it's working now? Already indexed files and folders cannot be removed by adding their name to a `.ppignore` file.

**Reply by gadelkareem** [reply]
I guess! I didn't edit the .ppignore file so it still has the christmas tree 🌲 😂

**Comment by gadelkareem** [comment]
There are ready libraries that are compatible with .gitignore. Maybe that's an easier solution?
https://github.com/sabhiram/go-gitignore/blob/master/ignore_ported_test.go#L107C18-L107C18
